src/python/grapl-tests-common/grapl_tests_common/clients/model_plugin_deployer_client.py

```diff
@@ -0,0 +1,67 @@
+"""
+Mostly copied from etc/local_grapl/bin/upload_plugin.py
+"""
+import os
+import sys
+import requests
+from pathlib import Path
+from http import HTTPStatus
+from typing import Dict
+
+_JSON_CONTENT_TYPE_HEADERS = {"Content-type": "application/json"}
+
+
+class ModelPluginDeployerException(Exception):
+    pass
+
+
+class ModelPluginDeployerClient:
+    def __init__(self, use_docker_links: bool = False) -> None:
+        hostname = "grapl-model-plugin-deployer" if use_docker_links else "localhost"
+        self.endpoint = f"http://{hostname}:8123"
+
+    def deploy(
+        self,
+        plugins_folder: Path,
+        jwt: str,
+    ) -> requests.Response:
+        paths = []
+        for subdir, _dirs, files in os.walk(plugins_folder):
+            for file in files:
+                if not self._is_valid_deployable_file(file):
+                    continue
+                paths.append(os.path.abspath(os.path.join(subdir, file)))
+
+        assert len(paths), "expected at least one file to be uploaded"
+
+        plugin_dict: Dict[str, str] = {}
+        for path in paths:
+            with open(path, "r") as f:
+                clean_path = str(path).split("model_plugins/")[-1]
+                contents = f.read()
+                if len(contents) == 0:
+                    # attempted hack: make __init__.py non empty
+                    # due to https://github.com/boto/botocore/pull/1328
+                    # otherwise the client just hangs
+                    assert clean_path.endswith("__init__.py")
```

Comment:
I feel like it wouldn't really be an error to have an empty file not called __init__.py ? Might be better to just debug log here if it isn't.

Reply:
legit